data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/coordinator/SourcePartitionStoreItem.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.source.coordinator;
+
+import java.time.Instant;
+
+/**
+ * The interface that should be implemented when creating store specific {@link org.opensearch.dataprepper.model.source.SourceCoordinationStore}
+ * plugin implementations.
+ * @since 2.2
+ */
+public interface SourcePartitionStoreItem {
+    String getSourcePartitionKey();
+    String getPartitionOwner();
+    String getPartitionProgressState();
+    SourcePartitionStatus getSourcePartitionStatus();
+    Instant getPartitionOwnershipTimeout();
+    Instant getReOpenAt();
+    Long getClosedCount();
```

Comment:
Question: What is the closed count used for?

Reply:
The `closedCount` will be used to track the number of times that a partition has been closed. For example, if we are reading from an index in an OpenSearch cluster, we will read that index once, mark it as closed, and then at some point in the future (based on the `reOpenAt` timestamp), we will allow the partition to be processed again until we hit some configured max closedCount. The closedCount itself keeps us from infinitely pulling from the index (we need some way to mark the partition as COMPLETED)